rails_programming/hotwire/turbo.md

```diff
@@ -0,0 +1,494 @@
+
+A basic Turbo Frame, using Rails helpers, may look like so:
+
+~~~erb
+<%= turbo_frame_tag "article" do %>
+  Some content
+<% end %>
+~~~
+
+which will generate:
+
+~~~html
+<turbo-frame id="article">
+  Some content
+</turbo-frame>
+~~~
+
+Super easy! Note that the frames have an ID. The ID is how Turbo is able to identify a frame to find out which one is which.
+With the Turbo Frame helper, you can substitute the ID for a variable. For instance:
+
+~~~erb
+<% @articles.each do |article| %>
+  <%= turbo_frame_tag article %>
+    <%= article.title %>
+  <% end %>
+<% end %>
+~~~
+
+The above example will generate a turbo frame for every article. Each frame will have a unique id like `article_1` or `article_2` and all we had to include was our article variable! 
+
+#### Connecting to Other Frames
+
+Now that we have our first frame, we need to connect it to another frame to swap the content out! This is very simple, all we have to do is create a link inside of a Turbo Frame where the requested view *also* has a Turbo Frame with the **same ID**.
+
+Lets connect the `/show` and `/edit` pages of an Article!
+
+~~~erb
+# views/articles/show.html.erb
+
+...
+<%= turbo_frame_tag @article %>
+  Content for our article!
+  <%= link_to "Edit Article", edit_article_path(@article) %>
+<% end %>
+...
+~~~
+
+~~~erb
+# views/articles/edit.html.erb
+
+...
+<%= turbo_frame_tag @article %>
+  Form to edit the article
+  <%= link_to "Return to Article", @article %>
+<% end %>
+...
+~~~
+
+That's all we have to do! Turbo will recognize that our destination URL, the `/show` or `/edit` page, has a matching Turbo Frame and will replace the frame region with the content from the new page's frame! Something else to note is that this does work with forms as well. In our controller, if the `update` action contains `redirect_to @article`, then our Turbo Frame will be updated when we submit our form just like if we clicked a link.
+
+Now that we have frames connected, what about the things outside of the frame? Anything outside of the frame does not change. If we were going from `/show` to `/edit`, then the content outside of the frame would still be the same content of the `/show` page and we would not receive any content from outside of the `/edit` frame either. The current page also does not change. We still stay on the `/show` page and if we were to refresh, we would still be here. We did not navigate to a new page, we only selected content from our destination and inserted it into our current page! (But using our previous knowledge of Turbo Drive, we can advance our history using `data-turbo-action` if we desire.)
+
+#### Breaking out of a Turbo Frame
+
+Sometimes you may have a link inside of the Turbo Frame that you want to act as a normal page navigation. To do so, just add `data-turbo-frame="_top"` to the element. An example with a Rails link helper:
```

Comment:
`Just` a general remark, you used `just`, `simply` and similar expressions in some places. There has been some discussion
around that, that such things may discourage people who have a hard time implementing something, see this link that explains it better, consider scanning the text for such words and removing them, usually the reading becomes clearer and does not impose the feeling that things actually should feel simple once you try this stuff out: https://justsimply.dev